docs/decisions/0005-credly-xblock.rst

```diff
@@ -0,0 +1,116 @@
+-  criteria: Text to describe how one earns this badge.
+-  description: Text description of this badge.
+-  section_title: This should be the display name of the sub-section you
+   want to check the score from.
+-  pass_mark: Minimum grade required to award this badge.
+-  received_award: Boolean to store if the user received a badge for
+   this sub-section.
+-  check_earned: True if they are eligible for a badge.
+-  assertion_url: The URL of the asserted badge in case it was already
+   awarded.
+-  award_message: Message the user will see upon receiving a badge.
+-  motivation_message: Message the user will see if they do not qualify
+   for a badge.
+
+Once setup, this XBlock will display a button with the text “Click here
+to view your results.”, once the learner clicks on it, the JavaScript
+function getGrades will be executed to request the current grading of
+the setup sub-section, if the minimum pass mark is achieved, it will
+execute the method new_award_badge using the XBlock handlerUrl runtime
+to award a new badge using the current setup badging service set up on
+the edx-platform, and reload the XBlock HTML to display the newly
+created badge for this user, otherwise, it will reload the HTML to an
+error message.
+
+XBlock Proposal
+---------------
+
+Given how badgr-xblock can interact with Badgr and the edx-platform to
+award badges, we could create a new XBlock to replicate this exact
+functionality using Credly has a backend service for badging to award
+badges on sub-section completion. A template for this XBlock could be
+generated using the XBlock SDK, this SDK also contains various examples
+of how we could use XBlock for various scenarios.
+
+We could query course grades using the grades v1 API grade book
+endpoint, which will return a list of the scores for each graded
+subsection per user, another option could be using the XBlock get_parent
+to trace the parent block to get a problem using an usage_key setup on
+this XBlock, to award badges for a specific unit or problem, the
+disadvantage of this method would be that the XBlock can only get the
+grade of the unit or problem at the moment this XBlock is being
+rendered. A mix of these methods could also be used to award a badge
+based on the completion of a group of sub-sections or a group of
+usage_keys which are graded problems.
+
+As described before, there are various methods we could use to retrieve
+information from the XBlock or platform to determine the conditions to
+generate a badge assertion (award a badge), either by requesting data
+using any API endpoint on the edx-platform using JavaScript, accessing
+the environment using the XBlock runtime API, using information stored
+in any of the scopes of the XBlock, using information stored in the
+XBlock fields, or using any of the XBlock services, such as the user
+service. More information on what API methods are available can be found
+in the XBlock API guide.
+
```

Comment:
Could you please add a section to point out the pros/cons?.  I see these ones but feel free to delete/add  some/more:

**Cons:** 
- Course/subsection scope which means there would be no badges which involve multiple courses, for example courses of program.

**Pros:**
- Xblock in studio offers a UI where content creators can configure the Badge. Platform is missing this part, and the only way to customize a Badge is trough Django admin.
- If course does not require a Badge, then there is no need to set up the xblock for this course. Platform Course Completion has this issue, All courses that issue certificates are going to issue badges as well, therefore when using Badges there is a need to disable this feature per course in case badges are not required.
- It does not care if Course is in verified mode. As Platform Badging system relies on certificates, the platform can only award badges for people who purchased the verified track.